cobalt/network/network_module.cc

```diff
@@ -161,8 +162,50 @@ void NetworkModule::SetEnableHttp3FromPersistentSettings() {
   }
 }
 
-void NetworkModule::SetProtocolFilterUpdatePending() {
+bool NetworkModule::SetHttpProtocolFilterPersistentSetting(
+    const std::string& raw_json) {
+  base::Value old_config_json;
+  options_.persistent_settings->Get(network::kProtocolFilterKey,
+                                    &old_config_json);
+
+  if (raw_json.empty() &&
+      (!old_config_json.is_string() || !old_config_json.GetString().empty())) {
+    options_.persistent_settings->Set(network::kProtocolFilterKey,
+                                      base::Value());
+    protocol_filter_update_pending_ = true;
+    return true;
+  }
+
+  absl::optional<base::Value> old_config;
+  if (old_config_json.is_string()) {
+    old_config = base::JSONReader::Read(old_config_json.GetString());
+  }
+  absl::optional<base::Value> raw_config = base::JSONReader::Read(raw_json);
+  if (!raw_config) return false;
+  base::Value::List new_config;
+  if (!raw_config->is_list()) return false;
+  for (auto& filter_value : raw_config->GetList()) {
+    if (!filter_value.is_dict()) continue;
+    const auto& dict = filter_value.GetDict();
+    const base::Value* origin = dict.Find("origin");
+    const base::Value* alt_svc = dict.Find("altSvc");
+    if (!origin || !alt_svc) continue;
+    if (!origin->is_string() || !alt_svc->is_string()) continue;
+    base::Value::Dict dest_dict;
+    dest_dict.Set("origin", origin->GetString());
+    dest_dict.Set("altSvc", alt_svc->GetString());
+    new_config.Append(std::move(dest_dict));
+  }
+  if (new_config.empty()) return false;
```

Comment:
Can you also fix that this will only return false if the old config was also empty, and true otherwise?